applications/GeoMechanicsApplication/custom_conditions/U_Pw_face_load_interface_condition.cpp

```diff
@@ -190,26 +190,24 @@ void UPwFaceLoadInterfaceCondition<3,4>::CheckJointWidth(double& rJointWidth, bo
                                                             const double& MinimumJointWidth, const Element::GeometryType& Geom)
 {
     //Quadrilateral_interface_3d_4
-    array_1d<double, 3> pmid0;
-    array_1d<double, 3> pmid1;
-    array_1d<double,3> P2 = Geom.GetPoint( 2 );
-    noalias(pmid0) = 0.5 * (Geom.GetPoint( 0 ) + Geom.GetPoint( 3 ));
-    noalias(pmid1) = 0.5 * (Geom.GetPoint( 1 ) + P2);
+    const auto& P2 = Geom.GetPoint( 2 );
+    const auto& pmid0 = 0.5 * (Geom.GetPoint( 0 ) + Geom.GetPoint( 3 ));
+    const auto& pmid1 = 0.5 * (Geom.GetPoint( 1 ) + P2);
```

Comment:
Very minor comment, but we try to name reference variables with an `r` in front. We also use `snake_case` for local variables. You can find some guidelines in our [style guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide).

Reply:
```cpp
const auto& pmid0 = 0.5 * (Geom.GetPoint( 0 ) + Geom.GetPoint( 3 ));
```

I was getting ready to bash on about `pmid0` immediately becoming a dangling ref, but TIL that this is actually legal. Thanks!

[Here](https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/) are some details if anyone's wondering what I'm going on about:.

Reply:
Even though it is legal I prefer the variation without &. It is easier to understand and reason about. Computational wise they are the same.
The explanation in the link is correct but it is an exception in the language (see: https://en.cppreference.com/w/cpp/language/lifetime )
However, not everybody knows those exceptions.

Reply:
> Even though it is legal I prefer the variation without &

same here

Reply:
Agree.

Reply:
> Very minor comment, but we try to name reference variables with an `r` in front. We also use `snake_case` for local variables. You can find some guidelines in our [style guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide).

Thank you for the comment. I have renamed the variable according to your suggestion. I would like to add that we may consider to use a clang format file to aid contributors in following Kratos' Style Guide. If you feel that is a good suggestion I'm willing to help set that up.

Reply:
That's a good idea!

Reply:
> ```c++
> const auto& pmid0 = 0.5 * (Geom.GetPoint( 0 ) + Geom.GetPoint( 3 ));
> ```
> 
> I was getting ready to bash on about `pmid0` immediately becoming a dangling ref, but TIL that this is actually legal. Thanks!
> 
> [Here](https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/) are some details if anyone's wondering what I'm going on about.
> 
> So regarding the naming @rubenzorrilla, despite what the declaration would imply, that variable acts like a value would so I'd keep the name.

Thanks for not bashing, but following a more constructive approach. I appreciate that. Bashing is not beneficiary in any way. Constructive criticism is.

Reply:
> Even though it is legal I prefer the variation without &. It is easier to understand and reason about. Computational wise they are the same. The explanation in the link is correct but it is an exception in the language (see: https://en.cppreference.com/w/cpp/language/lifetime ) However, not everybody knows those exceptions.

I was not aware that the C++ Core Guidelines state that binding a reference-to-const should be an exception. Thanks for pointing that out. After our internal discussion I feel that the modified code is better. Thanks for your contribution.